jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1603,6 +1604,73 @@ ecma_builtin_typedarray_prototype_find_index (ecma_value_t this_arg, /**< this a
   return ecma_builtin_typedarray_prototype_find_helper (this_arg, predicate, predicate_this_arg, false);
 } /* ecma_builtin_typedarray_prototype_find_index */
 
+/**
+ * The %TypedArray%.prototype object's 'indexOf' routine
+ *
+ * See also:
+ *          ECMA-262 v6, 22.2.3.13
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_bultin_typedarray_prototype_index_of (ecma_value_t this_arg, /**< this argument */
+                                           ecma_value_t arg1, /**< searchElement */
+                                           ecma_value_t arg2) /**< fromIndex */
+{
+  if (!ecma_is_typedarray (this_arg))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Argument 'this' is not a TypedArray."));
+  }
+
+  ecma_object_t *typedarray_p = ecma_get_object_from_value (this_arg);
+  uint32_t length = ecma_typedarray_get_length (typedarray_p);
+
+  if (length == 0)
+  {
+    return ecma_make_integer_value (-1);
+  }
+
+  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+
+  ECMA_OP_TO_NUMBER_TRY_CATCH (arg_from_idx, arg2, ret_value);
```

Comment:
Could you please the `ecma_get_number` operation directly instead of the `ECMA_OP_TO_NUMBER_TRY_CATCH`?